### PR TITLE
Tooltip for forms

### PR DESCRIPTION
**Pull request by louismaximepiton**
Adding tooltips in forms, based on [design system example](https://system.design.orange.com/0c1af118d/p/20500e-form/b/16bb53)
Waiting for a discussion between a11y, design and us to know which implementation to have.

### Other examples
For posterity, here is what it looks like aside an input:

```html
{{< example >}}
<div class="mb-3">
  <label for="tooltipTextInput2" class="form-label">Tooltip input</label>
  <div class="d-flex gap-2 align-items-center">
    <input type="text" id="tooltipTextInput2" class="form-control" placeholder="Tooltip input">
    <button type="button" class="form-helper" data-bs-toggle="popover" data-bs-placement="top" data-bs-content="Help for input">
      <span class="visually-hidden">Helper for input</span>
    </button>
  </div>
</div>
<div>
  <label for="tooltipSelect2" class="form-label is-required">Tooltip select menu</label>
  <div class="d-flex gap-2 align-items-center">
    <select id="tooltipSelect2" class="form-select" required>
      <option value="">Tooltip select</option>
    </select>
    <button type="button" class="form-helper" data-bs-toggle="tooltip" data-bs-placement="top" data-bs-title="Help for select menu">
      <span class="visually-hidden">Helper for select menu</span>
    </button>
  </div>
</div>
{{< /example >}}
```

### CODE REVIEW

**Review by MewenLeHo — APPROVED** [review]
🆗

**Comment by julien-deramond** [comment]
Haven't looked at the details but IMO we should use popovers and not tooltips.

**Comment by louismaximepiton** [comment]
Adding what was said in design spec meeting:
- Might be a popover
- But we should display both implementation (tooltip, popover)

File: scss/forms/_labels.scss [hunk 1]
```diff
@@ -21,6 +21,21 @@
   color: $accessible-orange;
   content: "*";
 }
+
+.btn-helper {
```
Comment (julien-deramond) [inline scss/forms/_labels.scss:25]: Doesn't it make sense to rather call it `.form-helper`?

File: scss/forms/_labels.scss [hunk 2]
```diff
@@ -21,6 +21,21 @@
...
+  display: inline-block;
+  flex-shrink: 0; // Defensive CSS
```
Comment (julien-deramond) [inline scss/forms/_labels.scss:27]: You can remove this comment

File: site/content/docs/5.2/forms/overview.md [hunk 3]
```diff
@@ -160,6 +160,67 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
 </form>
 {{< /example >}}
 
+## Form helper
+
+Provide some help on an input with `.btn-helper`. **Be sure to include text for screen readers**, as we've done with `<span class="visually-hidden">`.
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:165]:
```suggestion
Form helper is a button using `.form-helper` that provides extra help information to a form control. Be sure to include text for screen readers with a visually hidden text.
```

File: site/content/docs/5.2/forms/overview.md [hunk 4]
```diff
@@ -160,6 +160,67 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
...
+
+On top on this button, apply a [popover]({{< docsref "/components/popovers" >}}) to display the help. A [tooltip]({{< docsref "/components/tooltips" >}}) can also be applied for small and simple text helpers without any link, button, etc...
+
+{{< example >}}
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:169]: I don't have any ideas right now but it is very strange to display those 2 exact same buttons like this. On top of that, they have a pre-defined margin-left which make them look weird.
I'm wondering if we couldn't just keep the real-life examples.

File: site/content/docs/5.2/forms/overview.md [hunk 5]
```diff
@@ -160,6 +160,67 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
...
+<button type="button" class="btn-helper" data-bs-toggle="tooltip" data-bs-placement="top" data-bs-title="Help with tooltip.">
+  <span class="visually-hidden">Helper for input</span>
+</button>
+{{< /example >}}
+
+The helper button should be displayed right after a label.
+
+{{< example >}}
+<div class="mb-3">
+  <label for="tooltipTextInput" class="form-label">Tooltip input</label>
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:182]: This one is called "Tooltip" whereas it is a popover

File: site/content/docs/5.2/forms/overview.md [hunk 6]
```diff
@@ -160,6 +160,67 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
...
+  <button type="button" class="btn-helper" data-bs-toggle="tooltip" data-bs-placement="top" data-bs-title="Help for select menu">
+    <span class="visually-hidden">Helper for select menu</span>
+  </button>
+  <select id="tooltipSelect" class="form-select" required>
+    <option value="">Tooltip select</option>
+  </select>
+</div>
+{{< /example >}}
+
+<!-- To remove just here to show that it could be well and easily integrated everywhere -->
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:199]: You can remove this part and include it in the description for the future so we don't forget it

**Review by julien-deramond — COMMENTED** [review]
General feedback

**Comment by julien-deramond** [comment]
I've added a migration guide note via https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1444/commits/e858109925a043e382543581e3783bd735029adf. I let you check that @louismaximepiton

File: scss/forms/_labels.scss [hunk 7]
```diff
@@ -21,6 +21,21 @@
   color: $accessible-orange;
   content: "*";
 }
+
+.form-helper {
```
Comment (julien-deramond) [inline scss/forms/_labels.scss:25]: (Firefox || Chrome) + macOS, there's an issue with the vertical alignment:

Screenshot 2022-11-30 at 13 01 09

File: site/content/docs/5.2/forms/overview.md [hunk 8]
```diff
@@ -164,6 +164,33 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
 </form>
 {{< /example >}}
 
+## Form helper
+
+Form helper is a button using `.form-helper` that provides extra help information to a form control. Be sure to include text for screen readers with a visually hidden text.
+
+On top of this button, apply a [popover]({{< docsref "/components/popovers" >}}) to display the help. A [tooltip]({{< docsref "/components/tooltips" >}}) can also be applied for small and simple text helpers without any actionable element inside.
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:171]: Wording suggestion:

```suggestion
Depending on the extra help information content, choose wisely between triggering a [popover]({{< docsref "/components/popovers" >}}) or a [tooltip]({{< docsref "/components/tooltips" >}}). Tooltips should be used for small and simple texts without any actionable elements inside, while popovers are used in other cases.
```

File: site/content/docs/5.2/forms/overview.md [hunk 9]
```diff
@@ -164,6 +164,33 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
...
+
+The helper button should be displayed right after the label.
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:173]:
```suggestion
This form helper should be displayed right after the label.
```

File: site/content/docs/5.2/forms/overview.md [hunk 10]
```diff
@@ -164,6 +164,33 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
...
+
+{{< example >}}
+<div class="mb-3">
+  <label for="tooltipTextInput" class="form-label">Input with helper inside a Popover</label>
+  <button type="button" class="form-helper" data-bs-toggle="popover" data-bs-placement="top" data-bs-content="Help for input">
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:178]: In terms of accessibility, there's no need to "link" this button that provides extra information to the associated label and/or input?

**Review by julien-deramond — CHANGES_REQUESTED** [review]
This PR will need the validation from:
* Designers mainly for the explanation of the use of both tooltips/popovers
* A11y team to show them this new usage, check if some things could be added to improve a11y (even by knowing that popovers and tooltips are not yet fully accessible-other topic to tackle later on).

File: scss/_variables.scss [hunk 11]
```diff
@@ -968,6 +969,12 @@ $form-label-disabled-color:             $gray-500 !default; // Boosted mod
 $form-label-required-margin-left:       .1875rem !default; // Boosted mod
 // scss-docs-end form-label-variables
 
+// scss-docs-start form-helper-variables
+$form-helper-size:                      1.125rem !default; // Boosted mod
+$form-label-helper-margin-bottom:       .3125rem !default; // Boosted mod
```
[the remark this answers is not in the document]
Comment (louismaximepiton) [reply scss/_variables.scss:974]: All the other one presenetd in this page have such a paragraph.

File: site/content/docs/5.2/forms/overview.md (hunk 10 above)
Comment (louismaximepiton) [reply site/content/docs/5.2/forms/overview.md]: FYI, a11y said that there's no need to add some aria selectors in here. Just the behavior of the tooltips/popovers need to be checked and changed here.

**Comment by louismaximepiton** [comment]
Quick design review from Franco: it looks fine for the spacing and the size of the input. The only thing to change here is the tooltip/popover look and maybe the blue since some work has been done on design tokens (IMO, it should be tackled with dark mode support coming from Bootstrap soon).

File: site/content/docs/5.2/forms/overview.md [hunk 12]
```diff
@@ -164,6 +164,30 @@ Use the `.is-required` class on `label` to add a `*` symbol after it.
...
+This form helper should be displayed right after the label.
+
+{{< example >}}
+<label for="tooltipTextInput" class="form-label">Input with helper inside a Popover</label>
+<button type="button" class="form-helper" data-bs-toggle="popover" data-bs-placement="top" data-bs-content="Help for input">
+  <span class="visually-hidden">Helper for input</span>
+</button>
+<input type="text" id="tooltipTextInput" class="form-control" placeholder="Tooltip input">
+
+<label for="tooltipSelect" class="form-label is-required mt-3">Select menu with helper inside a Tooltip</label>
```
Comment (julien-deramond) [inline site/content/docs/5.2/forms/overview.md:182]: Seeing this `.mt-3` here, I'm wondering if we shouldn't do the following:
```html
<div class="mb-3">
   <!-- First example -->
</div>
<div>
   <!-- Second example -->
</div>
```

to have consistency with the other forms in the docs. But might be dumb. You'll tell me.

File: site/content/docs/5.2/migration.md [hunk 13]
```diff
@@ -26,6 +26,22 @@ If you need more details about the changes, please refer to the [v5.2.3 release]
...
+- <span class="badge bg-success">New</span> Form helpers icon buttons will help you to provide extra help information with the `.form-helper` utility class.
+
+### CSS and Sass variables
+
+- <details class="mb-2">
+    <summary><span class="badge bg-success">New</span> Sass variables:</summary>
+    <ul>
+      <li><code>$form-helper-size</code></li>
+      <li><code>$form-helper-label-margin-bottom</code></li>
+      <li><code>$form-helper-label-margin-left</code></li>
```
Comment (julien-deramond) [inline site/content/docs/5.2/migration.md:40]: This one can be removed isn't it?

**Review by julien-deramond — APPROVED** [review]
LGTM!

**Pull request merged**